Components/Button/Button.tsx

```diff
@@ -0,0 +1,107 @@
+import * as React from 'react';
+import styled from 'styled-components';
+import { Link } from '../Link/Link';
+
+export interface Style {
+    background?: string,
+    border?: string,
+    color?: string,
```

Comment:
I'd rather we go down the path of defining class names versus huge `style` blocks in the JS.  This way we're not tying our logic to our styles since they're different problems to solve.

Reply:
@NickGlenn Writing CSS in JS is necessary for UI kits like this... attempting to bring style sheets in from node_modules is a pain in the ass. Additionally we would then have 2 files to manage in the library instead of just one. I think that providing a default style set is a perfect use case for CSS in JS... This also allows easily overwritten styles when customizing from outside of the library.

Reply:
@canfie1d I know it's a pain to deal with CSS loading from `node_modules/`, however, one solution (since we're using either SCSS or Stylus) is that we could expose a plugin for one of those architectures - we did this with [helpful-ui](https://npmjs.com/package/helpful-ui) and I'm sure we could put our heads together and find a clean solution.  That way we can avoid the end-user's maintenance nightmare of dealing with styling in 2 different domains.